ua/encode.go

```diff
@@ -78,18 +79,25 @@ func encode(val reflect.Value, name string) ([]byte, error) {
 			if val.IsNil() {
 				return nil, nil
 			}
-			return encode(val.Elem(), name)
+			return logIntercept(encode(val.Elem(), name))
 		case reflect.Struct:
-			return writeStruct(val, name)
+			return logIntercept(writeStruct(val, name))
 		case reflect.Slice:
-			return writeSlice(val, name)
+			return logIntercept(writeSlice(val, name))
 		case reflect.Array:
-			return writeArray(val, name)
+			return logIntercept(writeArray(val, name))
 		default:
 			return nil, errors.Errorf("unsupported type: %s", val.Type())
 		}
 	}
-	return buf.Bytes(), buf.Error()
+	return logIntercept(buf.Bytes(), buf.Error())
+}
+
+func logIntercept(bytes []byte, err error) ([]byte, error) {
+	if debugCodec {
+		fmt.Printf("Wrote:\n%s\n", hex.Dump(bytes))
```

Comment:
Do we have to split the output over two lines? Can you provide an example output? The other debug output uses `encode: ` as prefix. Let's stick with that.

```go
fmt.Printf("encode: wrote %s\n", hex.Dump(b))
```

Reply:
Yes it is necessary otherwise the first line in indent and that looks bad
At the moment it looks like this:
```
Wrote:
00000000  02 00 00 f3 4d 51 19 77  4b c6 46 40 d7 d9 01 08  |....MQ.wK.F@....|
00000010  00 00 00 00 00 00 00 ff  ff ff ff 10 27 00 00 00  |............'...|
00000020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 05 00  |................|
00000030  00 00 c0 55 02 00 00 00  ff ff ff ff 00 00 ff ff  |...U............|
00000040  ff ff c0 55 03 00 00 00  ff ff ff ff 00 00 ff ff  |...U............|
00000050  ff ff c0 55 05 00 00 00  ff ff ff ff 00 00 ff ff  |...U............|
00000060  ff ff c0 55 11 00 00 00  ff ff ff ff 00 00 ff ff  |...U............|
00000070  ff ff c0 55 0e 00 00 00  ff ff ff ff 00 00 ff ff  |...U............|
00000080  ff ff                                             |..|

```

Reply:
the last `\n` could be omitted though as the dump itself is terminated by a `\n`